### PR TITLE
chore: angular 18

### PR DESCRIPTION
**Pull request by sdo-1A**
## Proposed change

Migration to Angular 18

### CODE REVIEW

File: packages/@o3r/workspace/package.json [hunk 1]
```diff
@@ -106,13 +107,13 @@
...
-    "@ngrx/effects": "~17.2.0",
-    "@ngrx/store-devtools": "~17.2.0"
+    "@angular/material": "~18.0.0",
+    "@ngrx/router-store": "~18.0.0",
+    "@ngrx/effects": "~18.0.0",
+    "@ngrx/store-devtools": "~18.0.0"
   },
   "engines": {
-    "node": ">=18.0.0"
+    "node": "^18.19.1 || ^20.11.1 || >=22.0.0"
```
Comment (vscaiceanu-1a) [inline packages/@o3r/workspace/package.json:116]: `^18.19.1 \|\| ^20.11.1 \|\| ^22.0.0`

File: packages/@o3r/components/src/rules-engine/placeholder.rules-engine.effect.spec.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import {TestBed} from '@angular/core/testing';
 import {provideMockActions} from '@ngrx/effects/testing';
-import type {TypedAction} from '@ngrx/store/src/models';
+import type {Action} from '@ngrx/store/src/models';
```
Comment (matthieu-crouzet) [inline packages/@o3r/components/src/rules-engine/placeholder.rules-engine.effect.spec.ts:3]:
```suggestion
import type {Action} from '@ngrx/store';
```

File: packages/@o3r/workspace/package.json (hunk 1 above)
Comment (fpaul-1A) [reply packages/@o3r/workspace/package.json:116]: is that needed ?
we are using the same as in angular repo https://github.com/angular/angular/blob/main/packages/core/package.json#L8

Comment (vscaiceanu-1a) [reply packages/@o3r/workspace/package.json:116]: According to their docs: https://angular.dev/reference/versions#actively-supported-versions
But maybe for some reason they do not want to block future majors. So not blocking.

**Review by vscaiceanu-1a — APPROVED** [review]

File: packages/@o3r/workspace/package.json (hunk 1 above)
Comment (fpaul-1A) [reply packages/@o3r/workspace/package.json:116]: Oh sorry, I didn't noticed the `^22` vs `>=22` I was too focused on the `\|\|` :D
I'm not sure why the mismatch, but that means we are pretending to support the odd versions of Node after 22
I suggest we keep it aligned with the package.json of angular for now but keep an eye on it for future releases of Node and Angular

**Review by matthieu-crouzet — APPROVED** [review]

**Review by mrednic-1A — APPROVED** [review]

**Pull request merged**